snuba/admin/views.py

```diff
@@ -390,20 +393,26 @@ def clickhouse_system_query() -> Response:
         port = req["port"]
         storage = req["storage"]
         raw_sql = req["sql"]
+        sudo_mode = req.get("sudo", False)
     except KeyError:
         return make_response(jsonify({"error": "Invalid request"}), 400)
 
     try:
-        result = run_system_query_on_host_with_sql(host, port, storage, raw_sql)
+        result = run_system_query_on_host_with_sql(
+            host, port, storage, raw_sql, sudo_mode, g.user
+        )
         rows = []
         rows, columns = cast(List[List[str]], result.results), result.meta
 
-        if columns:
+        if columns is not None:
```

Comment:
I am not sure about this change. Specifically I don't know if testing `if columns` rather than nullability was intentional. If yes what was the scenario ?

Reply:
Can't speak for the intention but I think `if columns` would cover both cases like `columns is not None` and `len(columns) != 0`. AFAIK, `columns` is most likely a `tuple`. Probably that it why it was added. My suggestion is to keep the original line as is to avoid surprises.

Reply:
The original would not work as, for system statements the columns list is empty.  Which makes sense as the response is not a table.

That's why I was wondering if there was a reason for raising an error in case of empty columns list. I cannot think about one.

Reply:
The result for the system queries tool often is a table:

```
SELECT * FROM system.mutations WHERE is_done=0
```

Usually when we're using this endpoint we are indeed returning a table

Reply:
So, changing should not be an issue,  right as it was not implemented this way "to" prevent non table results, but "because" results were tables ?